card/src/main/java/com/adyen/checkout/card/ui/CardNumberInput.kt

```diff
@@ -68,7 +67,7 @@ open class CardNumberInput @JvmOverloads constructor(
     private fun formatProcessedString(processedValue: String): String {
         val result =
             splitStringWithMask(processedValue, *if (isAmexCard) AMEX_CARD_NUMBER_MASK else DEFAULT_CARD_NUMBER_MASK)
-        return TextUtils.join(DIGIT_SEPARATOR.toString() + "", result).trim { it <= ' ' }
```

Comment:
Not related to this change but I think it would be nice to extract this formatting logic to somewhere else and unit test it.

Reply:
Agreed, can you create a ticket for it? 🙏